src/v3/src/transformer/uischema/transform.ts

```diff
@@ -55,9 +59,62 @@ export const setFocusOnFirstElement: TransformStepFn = (formbag) => {
   return formbag;
 };
 
+export const updateCustomFields: TransformStepFn = (formbag) => {
+  traverseLayout({
+    layout: formbag.uischema,
+    predicate: (el) => (el.type === 'Field'),
+    callback: (el) => {
+      const fieldElement = (el as FieldElement);
+      const { options: { inputMeta: { options } } } = fieldElement;
+      if (Array.isArray(options) && options[0]?.value) {
+        const [option] = options;
+        if (option.label === 'display') {
+          // TODO: OKTA-538689 Missing type in interface, have to cast to any
+          const input = (option.value as any)?.value;
+          fieldElement.options.format = input.inputType;
+          fieldElement.options.customOptions = input.options;
+          if (input.inputType === 'select' && input.format === 'country-code') {
+            const countryCodeObj = CountryUtil.getCountryCode();
+            const countryOptions = Object.entries(countryCodeObj).map(([code, label]) => ({
+              label,
+              value: code,
+            } as IdxOption));
+            fieldElement.options.customOptions = countryOptions;
+          } else if (input.inputType === 'text') {
+            // Text type that has options must remove options for renderers.tsx
+            // to map to correct element
+            fieldElement.options.inputMeta.options = undefined;
+          }
+        }
+      }
+    },
+  });
+  return formbag;
+};
+
+// TODO: temporary solution for custom fields in profile enrollment
```

Comment:
is there a ticket that will allow us to remove this?

Reply:
Will check with ODS team to get a jira for the wrapper work they are doing. 🙏